src/libmongoc/CMakeLists.txt

```diff
@@ -1058,6 +1079,10 @@ mongoc_add_test (test-libmongoc FALSE ${test-libmongoc-sources})
 mongoc_add_test (test-mongoc-gssapi FALSE ${PROJECT_SOURCE_DIR}/tests/test-mongoc-gssapi.c)
 mongoc_add_test (test-mongoc-cache FALSE ${PROJECT_SOURCE_DIR}/tests/test-mongoc-cache.c)
 
+if (TARGET get-csfle AND TARGET test-libmongoc)
+   target_compile_definitions (test-libmongoc PRIVATE "TESTING_CSFLE_OVERRIDE_PATH=\"$<TARGET_PROPERTY:get-csfle,CSFLE_PATH>\"")
```

Comment:
Is `get-csfle` expected to run in Evergreen? The ASAN `*-cse` task failures look unrelated (I think due to [`BSON_MEMCHECK`](http://mongoc.org/libbson/current/valgrind.html) shenanigans with libmongocrypt.). I do not see logs for the get-csfle task in [the VS 2017 task](https://spruce.mongodb.com/task/mongo_c_driver_windows_2017_test_latest_server_auth_sasl_openssl_cse_patch_28d66415d35bec28dba76c054bde1a0c9d69f56c_62606bcd32f41728ef1faa7c_22_04_20_20_23_43/logs?execution=0).

Reply:
Yes, it should execute by default. I'll have to inspect further. This will not run if `get-csfle` is not defined, which will not be defined unless `Python3::Interpreter` is defined. [In these build logs](https://spruce.mongodb.com/task/mongo_c_driver_windows_2017_debug_compile_sasl_openssl_cse_patch_28d66415d35bec28dba76c054bde1a0c9d69f56c_62606bcd32f41728ef1faa7c_22_04_20_20_23_43?execution=0) it says that `Python3` is not found, which means it is likely too old of a CMake version to have the `FindPython3.cmake` module.

We may or may not be okay with this, as the tests should always pass regardless of whether `csfle` is injected into them. Having some builds that use `mongocryptd` and some that use `csfle` may be useful, but we may want to make that more explicit.

Reply:
> We may or may not be okay with this, as the tests should always pass regardless of whether csfle is injected into them. Having some builds that use mongocryptd and some that use csfle may be useful, but we may want to make that more explicit.

Having at least one task use `csfle` and one task use `mongocryptd` seems desirable. I prefer it be made explicit. Otherwise, Evergreen host changes may change which is being used, and we may not notice that.

Perhaps a simple option is to add a cmake flag: `DISABLE_CSFLE_DOWNLOAD`? One task in Evergreen can set that flag to guarantee it uses mongocryptd?